### ISSUE TITLE
Feature: SMART_ACTION to disable onkill reputation gains and loot/gold drops for source creature

### ISSUE THREAD

**Issue by Gultask** [azerothcore/azerothcore-wotlk#16359]
### Describe your feature request or suggestion in detail

Title

### Additional context

Example is Shattered Halls.
In the entrance, the 2 possible entries there respawn constantly if the Legionnaire isn't killed, but they do not drop gold or loot or rep.
Another are the Assassins Kargath spawns, they are the same.

The same entry drops loot and gold and rep elsewhere in the dungeon

https://github.com/azerothcore/azerothcore-wotlk/issues/16359

**Comment by elthehablo** [comment]
Sounds useful. Probably also needed for more bosses in wotlk in some way or another

**Comment by Gultask** [comment]
Yeah. I can just use an addon (IdTip), but in this case it's sniffed.

**Comment by Gultask** [comment]
This might be possible with just messing with PlayerDamageReq. Gotta investigate further
